### PR TITLE
Classification-based evaluation using `LanguageModel`

### PR DESCRIPTION
**Pull request by junya-takayama**
Add `LLMLabel` and `ChatLLMLabel`.
Different from `LLMScore` and `ChatLLMScore`, they deal with labels on a non-Likert scale.
You can define label names and assign a score to each label. The mean score and the label distribution are output.

### CODE REVIEW

File: flexeval/core/metric/llm_label.py [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+        label_names: A list of valid label names.
+        label_points: A list of points for each label specified in label_names.
+        batch_size: The batch size for the evaluator.
+        disable_tqdm: Whether to disable the progress bar.
+        category_key: A key to create category-wise mean score.
+            The category key is expected to be in task inputs.
+
+    Examples:
+        >>> from flexeval_instruction.plugins.llm_label import LLMLabel
+        >>> from flexeval import OpenAIChatAPI, Jinja2PromptTemplate
```
Comment (ryokan0123) [inline flexeval/core/metric/llm_label.py:108]:
```suggestion
        >>> from flexeval import OpenAIChatAPI, Jinja2PromptTemplate, LLMLabel
```

File: flexeval/core/metric/llm_label.py [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+    Args:
+        language_model: An instance of `LanguageModel` to evaluate the output of the model.
+        prompt_template: An instance of `PromptTemplate` to embed the input for the evaluator.
+        label_names: A list of valid label names.
+        label_points: A list of points for each label specified in label_names.
+        system_message: A system message to be prepended to the input for the evaluator.
+        batch_size: The batch size for the evaluator.
+        disable_tqdm: Whether to disable the progress bar.
+        valid_score_range: A tuple of two integers representing the valid score range.
+            If the parsed score is out of the range, it will be ignored.
```
Comment (ryokan0123) [inline flexeval/core/metric/llm_label.py:243]:
```suggestion
```

File: flexeval/core/metric/llm_label.py [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+        >>> from flexeval import ChatLLMScore, OpenAIChatAPI, Jinja2PromptTemplate
+        >>> language_model = OpenAIChatAPI(model_name="gpt-3.5-turbo")
+        >>> template = "Evaluate the quality of this text on a scale of Good/Bad.\\n`{{ lm_output }}`\\nPut the label at the end like [[Good]]."
+        >>> prompt_template = Jinja2PromptTemplate(template)
+        >>> system_message = "This is the system message."
+        >>> label_names = ["Good", "Bad"]
+        >>> label_points = [1.0, 0.0]
+        >>> llm_label = ChatLLMLabel(language_model, prompt_template, label_names, label_points)
+        >>> lm_outputs = ["Hello, world!", "Good morning!"]
+        >>> result = llm_score.evaluate(lm_outputs)
```
Comment (ryokan0123) [inline flexeval/core/metric/llm_label.py:257]:
```suggestion
        >>> result = llm_label.evaluate(lm_outputs)
```

**Review by ryokan0123 — APPROVED** [review]
LGTM.
Just fixed a few typos in docstrings.

**Comment by junya-takayama** [comment]
Thanks for the correction!
修正漏れお恥ずかしい……🫣

**Pull request merged**